nixpkgs/nixpkgs.bzl

```diff
@@ -15,6 +15,16 @@ nixpkgs_git_repository = repository_rule(
   local = False,
 )
 
+def _mk_build_expression(ctx):
+  """Generate a nix expression that picks a package from nixpkgs.
+  """
+  if ctx.attr.attribute and ctx.attr.expression:
+    fail("'attribute' and 'expression' are mutually exclusive.")
```

Comment:
I don't think the two should be mutually exclusive. `-A` and `-E` are not mutually exclusive flags to `nix-build`. When used together, `-E e -A some.path` means `e.some.path`. We should reproduce the same semantics here.